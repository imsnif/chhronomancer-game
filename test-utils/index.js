module.exports = {
  mockFB () {
    const login = require('react-native-facebook-login')
    const FBLogin = jest.fn()
    FBLogin.mockReturnValue(null)
    login.FBLogin = FBLogin
    login.FBLoginManager = {LoginBehaviors: {Web: 2}}
    return FBLogin
  },
  navigatorMock (routeName) {
    return {
      getCurrentRoutes: () => [routeName],
      push: () => {}
    }
  }
}
