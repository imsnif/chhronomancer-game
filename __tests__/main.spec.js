import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import serverMock from '../server-mock'
import { mockFB } from '../test-utils'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

beforeEach(() => {
  jest.useFakeTimers()
  serverMock()
})

afterEach(() => {
  const clockStore = require('../stores/clock').default
  jest.useRealTimers()
  clockStore.destroy()
  jest.useFakeTimers()
})

test('Main component (with server mock) => renders correctly', () => {
  const Chronomancer = require('../components/main').default
  const statsStore = require('../stores/stats').default
  statsStore.changePlayerId(1)
  const tree = renderer.create(
    <Chronomancer />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Main component (with server mock) => renders correctly - Waiting For Game', () => {
  const Chronomancer = require('../components/main').default
  const statsStore = require('../stores/stats').default
  statsStore.connect()
  statsStore.changePlayerId('foo')
  const tree = renderer.create(
    <Chronomancer />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Main component (with server mock) => renders correctly - Login page', () => {
  jest.resetModules()
  const statsStore = require('../stores/stats').default
  const Chronomancer = require('../components/main').default
  const FBLogin = mockFB()
  statsStore.changePlayerId('foo')
  statsStore.disconnect()
  renderer.create(
    <Chronomancer />
  ).toJSON()
  expect(FBLogin.mock.calls).toMatchSnapshot()
})
