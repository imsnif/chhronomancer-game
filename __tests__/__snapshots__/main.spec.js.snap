// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component (with server mock) => renders correctly - Login page 1`] = `
Array [
  Array [
    Object {
      "buttonView": <View
        style={
          Object {
            "backgroundColor": "black",
            "height": "100%",
            "width": "100%",
          }
        }
      />,
      "loginBehavior": 2,
      "onLogin": [Function],
      "onLoginFound": [Function],
      "permissions": Array [
        "email",
        "user_friends",
      ],
      "style": Object {
        "marginBottom": 10,
      },
    },
    Object {},
    Object {
      "enqueueCallback": [Function],
      "enqueueCallbackInternal": [Function],
      "enqueueElementInternal": [Function],
      "enqueueForceUpdate": [Function],
      "enqueueReplaceState": [Function],
      "enqueueSetState": [Function],
      "isMounted": [Function],
      "validateCallback": [Function],
    },
  ],
]
`;

exports[`Main component (with server mock) => renders correctly - Waiting For Game 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "green",
        "fontFamily": "telegrama_raw",
      }
    }
  >
    Connected! Waiting for game... (playerId: 
    foo
    )
  </Text>
</View>
`;

exports[`Main component (with server mock) => renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      collapsable={false}
      onStartShouldSetResponderCapture={[Function]}
      pointerEvents="auto"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
              Object {
                "scaleX": 1,
              },
              Object {
                "scaleY": 1,
              },
              Object {
                "rotate": "0deg",
              },
              Object {
                "skewX": "0deg",
              },
              Object {
                "skewY": "0deg",
              },
            ],
          },
          null,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#141313",
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "column",
              "height": "15%",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#329932",
                "flex": 0,
                "fontFamily": "telegrama_raw",
                "fontSize": 27.546799160701053,
                "height": "20%",
                "includeFontPadding": false,
                "lineHeight": 22,
                "textAlign": "center",
                "textAlignVertical": "bottom",
              }
            }
          >
            Actions: 
            10
          </Text>
          <View
            style={
              Object {
                "alignSelf": "center",
                "flexDirection": "row",
                "height": "30%",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "color": "#329932",
                    "flex": 1,
                    "fontFamily": "telegrama_raw",
                    "fontSize": 27.546799160701053,
                    "height": 9,
                    "includeFontPadding": false,
                    "lineHeight": 9,
                    "textAlign": "center",
                    "textAlignVertical": "bottom",
                  }
                }
              >
                Character
              </Text>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "color": "#329932",
                    "flex": 1,
                    "fontFamily": "telegrama_raw",
                    "fontSize": 27.546799160701053,
                    "height": 9,
                    "includeFontPadding": false,
                    "lineHeight": 9,
                    "textAlign": "center",
                    "textAlignVertical": "bottom",
                  }
                }
              >
                Timelines
              </Text>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "color": "#329932",
                    "flex": 1,
                    "fontFamily": "telegrama_raw",
                    "fontSize": 27.546799160701053,
                    "height": 9,
                    "includeFontPadding": false,
                    "lineHeight": 9,
                    "textAlign": "center",
                    "textAlignVertical": "bottom",
                  }
                }
              >
                [ Feed (2) ]
              </Text>
            </View>
          </View>
          <Image
            source={1}
            style={
              Object {
                "height": "30%",
                "resizeMode": "stretch",
                "width": "100%",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 2,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#141313",
                "flexDirection": "column",
                "height": "100%",
                "paddingBottom": 5,
                "paddingTop": 5,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 2,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                }
              }
            >
              <RCTScrollView>
                <View>
                  <View
                    style={
                      Object {
                        "backgroundColor": "#329932",
                        "flexDirection": "column",
                        "height": 60,
                        "marginBottom": 5,
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                          "width": "100%",
                        }
                      }
                    >
                      <Image
                        source={
                          Object {
                            "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
                          }
                        }
                        style={
                          Object {
                            "borderColor": "black",
                            "borderWidth": 1,
                            "flex": 0,
                            "height": 26,
                            "margin": 1,
                            "resizeMode": "stretch",
                            "width": 26,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "backgroundColor": "green",
                              "height": "100%",
                              "opacity": 0.5,
                              "width": "100%",
                            }
                          }
                        />
                      </Image>
                      <View
                        style={
                          Object {
                            "flex": 1,
                            "flexDirection": "column",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          numberOfLines={1}
                          style={
                            Object {
                              "alignSelf": "flex-start",
                              "color": "#141313",
                              "flex": 1,
                              "fontFamily": "telegrama_raw",
                              "fontSize": 22.955665967250873,
                              "paddingBottom": 3,
                              "paddingLeft": 3,
                              "paddingRight": 5,
                              "textAlignVertical": "center",
                            }
                          }
                        >
                          Aram
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "flex": 1,
                            "flexDirection": "column",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          numberOfLines={1}
                          style={
                            Object {
                              "alignSelf": "center",
                              "color": "#141313",
                              "flex": 1,
                              "fontFamily": "telegrama_raw",
                              "fontSize": 22.955665967250873,
                              "textAlignVertical": "center",
                            }
                          }
                        >
                          Timeline 0-1
                        </Text>
                      </View>
                    </View>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      numberOfLines={1}
                      style={
                        Object {
                          "alignSelf": "flex-start",
                          "color": "#141313",
                          "flex": 1,
                          "fontFamily": "telegrama_raw",
                          "fontSize": 19.894910504950758,
                          "textAlignVertical": "center",
                        }
                      }
                    >
                      has travelled here
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      numberOfLines={1}
                      style={
                        Object {
                          "alignSelf": "flex-start",
                          "color": "#141313",
                          "flex": 1,
                          "fontFamily": "telegrama_raw",
                          "fontSize": 15.303777311500582,
                          "textAlignVertical": "bottom",
                        }
                      }
                    >
                      48 years
                       ago
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "backgroundColor": "#329932",
                        "flexDirection": "column",
                        "height": 60,
                        "marginBottom": 5,
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                          "width": "100%",
                        }
                      }
                    >
                      <Image
                        source={
                          Object {
                            "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
                          }
                        }
                        style={
                          Object {
                            "borderColor": "black",
                            "borderWidth": 1,
                            "flex": 0,
                            "height": 26,
                            "margin": 1,
                            "resizeMode": "stretch",
                            "width": 26,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "backgroundColor": "green",
                              "height": "100%",
                              "opacity": 0.5,
                              "width": "100%",
                            }
                          }
                        />
                      </Image>
                      <View
                        style={
                          Object {
                            "flex": 1,
                            "flexDirection": "column",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          numberOfLines={1}
                          style={
                            Object {
                              "alignSelf": "flex-start",
                              "color": "#141313",
                              "flex": 1,
                              "fontFamily": "telegrama_raw",
                              "fontSize": 22.955665967250873,
                              "paddingBottom": 3,
                              "paddingLeft": 3,
                              "paddingRight": 5,
                              "textAlignVertical": "center",
                            }
                          }
                        >
                          Aram
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "flex": 1,
                            "flexDirection": "column",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          numberOfLines={1}
                          style={
                            Object {
                              "alignSelf": "center",
                              "color": "#141313",
                              "flex": 1,
                              "fontFamily": "telegrama_raw",
                              "fontSize": 22.955665967250873,
                              "textAlignVertical": "center",
                            }
                          }
                        >
                          Timeline 0-1
                        </Text>
                      </View>
                    </View>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      numberOfLines={1}
                      style={
                        Object {
                          "alignSelf": "flex-start",
                          "color": "#141313",
                          "flex": 1,
                          "fontFamily": "telegrama_raw",
                          "fontSize": 19.894910504950758,
                          "textAlignVertical": "center",
                        }
                      }
                    >
                      is locking the timeline
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      numberOfLines={1}
                      style={
                        Object {
                          "alignSelf": "flex-start",
                          "color": "#141313",
                          "flex": 1,
                          "fontFamily": "telegrama_raw",
                          "fontSize": 15.303777311500582,
                          "textAlignVertical": "bottom",
                        }
                      }
                    >
                      48 years
                       ago
                    </Text>
                  </View>
                </View>
              </RCTScrollView>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
